### ISSUE TITLE
Using Coiled with Nebari when pip packages are included in the environment

### ISSUE THREAD

**Issue by maxrjones** [nebari-dev/nebari#2378]
### Context

We have successfully used Coiled clusters via Nebari in the past. However, we recently found that the disabling of pip in the later releases of Nebari causes issues when there are pip packages included in the conda environment. `coiled.cluster()` works fine when only conda packages are included in the environment. I expect the best solution would be to pass the docker image from conda-store into `coiled.cluster(..., container="...")`, but I am not sure how to enable Coiled to access the image.

More specifically, the issue arises in the creating wheels step of package sync:

```python
import coiled
cluster = coiled.Cluster(n_workers=2)
```

```
╭───────────── Package Sync for carbonplan ─────────────╮
│ Fetching latest package priorities ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0:00:00 │
│ Scanning 727 conda packages        ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0:00:02 │
│ Scanning 425 python packages       ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0:00:02 │
│ Running pip check                  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0:00:01 │
│ Validating environment             ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0:00:01 │
│ Creating wheel for altair          ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0:00:00 │
╰──────────────────────────────────────────────────────────────────────────────╯

---Wheel Build Log for altair---
ERROR: Could not find an activated virtualenv (required).
```

### Value and/or benefit

The combination of Nebari + Coiled provides users with the benefits of Nebari for persistent JupyterHub access and Coiled for the scaling and monitoring of Dask clusters.

**Comment by jrbourbeau** [comment]
For reference, we just pushed out a new `coiled` release that _should_ fix things even with the recent `PIP_REQUIRE_VIRTUALENV=True` change in nebari. @maxrjones mentioned he would try it out when he gets a chance

**Comment by maxrjones** [comment]
Thank you for your prompt help with this issue! I can confirm that the issue is resolved after specifying `coiled==1.16.0` within the pip section of the conda environment file 🎉
